packages/ember/src/components/card.gts

```diff
@@ -0,0 +1,45 @@
+import Component from '@glimmer/component';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+
+interface CardSignature {
+  Element: HTMLDivElement;
+  Args: {
+    // eslint-disable-next-line no-unused-vars
+    onClick?: (evt: MouseEvent) => unknown;
+  };
+  Blocks: {
+    'card-header': [];
+    'card-body': [];
+  };
+}
+
+export default class Card extends Component<CardSignature> {
+  @action
+  onClick(evt: MouseEvent) {
+    evt?.preventDefault();
+    evt?.stopPropagation();
+
+    this.args.onClick?.(evt);
+  }
+
+  <template>
+    <div
+      class="card p-4 shadow-sm"
+      role="button"
+      {{on "click" this.onClick}}
```

Comment:
Let's add a `isClickable` getter if an onClick action is provided:
```ts
get isClickable() {
  return Boolean(this.args.onClick);
}
```

Then we can conditionally add the pointer and modifier:
```suggestion
      role={{if this.isClickable "button"}}
      {{! @glint-expect-error }}
      {{(if this.isClickable (modifier on "click" this.onClick))}}
```

Reply:
Fixed